### ISSUE TITLE
bug: ion-button text not read by screen readers if content not loaded immediately

### ISSUE THREAD

**Issue by sjdrew** [ionic-team/ionic-framework#28570]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Text from any ion-button is not read by screen readers under google chrome (or Edge) if the button is part of any context that did not load immediately. (For example some content that loaded 100 ms later).

Tested with windows 11 narrator and JAWS.

Win 11 narrator just says "button"
JAWS says "unlabeled button zero"

aria-live tags did not make any difference. Standard buttons work fine.

arial-label="button text" does solve, but according to ionic docs that should not be required.

### Expected Behavior

Text between the ion-button tag should be read by screen readers.

### Steps to Reproduce

Create a simple blank app.

Use this content:

```
<ion-content class="ion-padding" *ngIf="ready">
    <button style="padding:10px;color:white;background-color: blue;">Standard Button Read OK</button>
    <br><br>
    <ion-button>Ion Button Not Read</ion-button>
</ion-content>
```

add a setTimeout to set this.ready=true after 100ms 

Then screen readers will read the standard button tag text ok, but will not read the ion-button text.

### Ionic Info

Ionic:

   Ionic CLI                     : 7.1.1 (/opt/homebrew/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 7.5.6
   @angular-devkit/build-angular : 17.0.2
   @angular-devkit/schematics    : 17.0.2
   @angular/cli                  : 17.0.2
   @ionic/angular-toolkit        : 9.0.0

Capacitor:

   Capacitor CLI      : 5.5.1
   @capacitor/android : not installed
   @capacitor/core    : 5.5.1
   @capacitor/ios     : not installed

Utility:

   cordova-res                          : 0.15.4
   native-run (update available: 2.0.0) : 1.7.4

System:

   NodeJS : v18.18.2 (/opt/homebrew/Cellar/node@18/18.18.2/bin/node)
   npm    : 9.8.1
   OS     : macOS Unknown

**Comment by sjdrew** [comment]
In my testing (mostly on Parallels on Mac running windows 11 chrome/edge) it is 100% broken on ion-button and 100% works on regular button.  For that reason I would assume that Chromium team would say it was an ionic bug.  But I can give it a try, as you noted it does work 100% on Firefox.

**Comment by liamdebeasi** [comment]
That would be great. Screen readers should be announcing slotted text, so I don't see anything in Ionic that would be causing this to break.
